lib/GameConstants.h

```diff
@@ -393,85 +432,118 @@ class FactionID : public BaseForID<FactionID, si32>
-
-		DWELL_LVL_1 = DWELL_FIRST,
-		DWELL_LVL_7 = DWELL_LAST,
+		DEFAULT       = -50,
+		NONE          = -1,
+		MAGES_GUILD_1 = 0,
+		MAGES_GUILD_2,
+		MAGES_GUILD_3,
+		MAGES_GUILD_4,
+		MAGES_GUILD_5,
+		TAVERN,
+		SHIPYARD,
+		FORT,
+		CITADEL,
+		CASTLE,
+		VILLAGE_HALL,
+		TOWN_HALL,
+		CITY_HALL,
+		CAPITOL,
+		MARKETPLACE,
+		RESOURCE_SILO,
+		BLACKSMITH,
+		SPECIAL_1,
+		HORDE_1,
+		HORDE_1_UPGR,
+		SHIP,
+		SPECIAL_2,
+		SPECIAL_3,
+		SPECIAL_4,
+		HORDE_2,
+		HORDE_2_UPGR,
+		GRAIL,
+		EXTRA_TOWN_HALL,
+		EXTRA_CITY_HALL,
+		EXTRA_CAPITOL,
+		DWELL_FIRST = 30,
+		DWELL_LVL_2,
+		DWELL_LVL_3,
+		DWELL_LVL_4,
+		DWELL_LVL_5,
+		DWELL_LVL_6,
+		DWELL_LAST     = 36,
+		DWELL_UP_FIRST = 37,
+		DWELL_LVL_2_UP,
+		DWELL_LVL_3_UP,
+		DWELL_LVL_4_UP,
+		DWELL_LVL_5_UP,
+		DWELL_LVL_6_UP,
+		DWELL_UP_LAST = 43,
+
+		DWELL_LVL_1    = DWELL_FIRST,
+		DWELL_LVL_7    = DWELL_LAST,
 		DWELL_LVL_1_UP = DWELL_UP_FIRST,
 		DWELL_LVL_7_UP = DWELL_UP_LAST,
 
-		//Special buildings for towns.
+		// Special buildings for towns.
 		LIGHTHOUSE  = SPECIAL_1,
-		STABLES     = SPECIAL_2, //Castle
+		STABLES     = SPECIAL_2, // Castle
 		BROTHERHOOD = SPECIAL_3,
 
 		MYSTIC_POND         = SPECIAL_1,
-		FOUNTAIN_OF_FORTUNE = SPECIAL_2, //Rampart
+		FOUNTAIN_OF_FORTUNE = SPECIAL_2, // Rampart
 		TREASURY            = SPECIAL_3,
 
 		ARTIFACT_MERCHANT = SPECIAL_1,
-		LOOKOUT_TOWER     = SPECIAL_2, //Tower
+		LOOKOUT_TOWER     = SPECIAL_2, // Tower
 		LIBRARY           = SPECIAL_3,
 		WALL_OF_KNOWLEDGE = SPECIAL_4,
 
 		STORMCLOUDS   = SPECIAL_2,
-		CASTLE_GATE   = SPECIAL_3, //Inferno
+		CASTLE_GATE   = SPECIAL_3, // Inferno
 		ORDER_OF_FIRE = SPECIAL_4,
 
 		COVER_OF_DARKNESS    = SPECIAL_1,
-		NECROMANCY_AMPLIFIER = SPECIAL_2, //Necropolis
+		NECROMANCY_AMPLIFIER = SPECIAL_2, // Necropolis
 		SKELETON_TRANSFORMER = SPECIAL_3,
 
-		//ARTIFACT_MERCHANT - same ID as in tower
+		// ARTIFACT_MERCHANT - same ID as in tower
 		MANA_VORTEX      = SPECIAL_2,
-		PORTAL_OF_SUMMON = SPECIAL_3, //Dungeon
+		PORTAL_OF_SUMMON = SPECIAL_3, // Dungeon
 		BATTLE_ACADEMY   = SPECIAL_4,
 
 		ESCAPE_TUNNEL     = SPECIAL_1,
-		FREELANCERS_GUILD = SPECIAL_2, //Stronghold
+		FREELANCERS_GUILD = SPECIAL_2, // Stronghold
 		BALLISTA_YARD     = SPECIAL_3,
 		HALL_OF_VALHALLA  = SPECIAL_4,
 
 		CAGE_OF_WARLORDS = SPECIAL_1,
 		GLYPHS_OF_FEAR   = SPECIAL_2, // Fortress
 		BLOOD_OBELISK    = SPECIAL_3,
 
-		//ARTIFACT_MERCHANT - same ID as in tower
+		// ARTIFACT_MERCHANT - same ID as in tower
 		MAGIC_UNIVERSITY = SPECIAL_2, // Conflux
 	};
 
-	BuildingID(EBuildingID _num = NONE) : num(_num)
-	{}
-
-	STRONG_INLINE
-	bool IsSpecialOrGrail() const
+	BuildingID(EBuildingID _num = NONE)
+		: num(_num)
 	{
-		return num == SPECIAL_1 || num == SPECIAL_2 || num == SPECIAL_3 || num == SPECIAL_4 || num == GRAIL;
 	}
 
+	STRONG_INLINE
+	bool IsSpecialOrGrail() const { return num == SPECIAL_1 || num == SPECIAL_2 || num == SPECIAL_3 || num == SPECIAL_4 || num == GRAIL; }

```

Comment:
looks too long for a one-liner

Reply:
initial looks better

Reply:
Haven't noticed that it turned entire method into one-liner. Will look into it.

Reply:
Also I am against heavily indented things)